30413/HulpeMiruna/5.3/main.c

```diff
@@ -0,0 +1,209 @@
+    }
+}
+NodeChar *insertee (char *c,NodeChar *tree)
+{
+ if(tree==NULL)
+ {
+     NodeChar *p=(NodeChar *) malloc (sizeof(NodeT));
+     p->name=(char *) malloc (sizeof(char)*MAXNAME);
+     strcpy(p->name,c);
+     p->left=NULL;
+     p->right=NULL;
+     return p;
+ }
+ else{
+          if(strcmp(tree->name,c)<0)
+           {
+            tree->right=insertee (c,tree->right);
+           }
+           else tree->left=insertee (c,tree->left);
+        return tree;
+    }
+}
+int returnDate()
+{   int date;
+    int day,month,year;
+    time_t now=time(&now);
+    struct tm *local=localtime(&now);
+    day=local->tm_mday;
+    month=local->tm_mon+1;
+    year=local->tm_year+1900;
+    date = year*10000+month*100+day;
+    return date;
+}
+NodeChar *Expired(NodeT *tree,int level)
+{ NodeChar *T;
+if(level==0)
+{
+    T=NULL;
+}
+     if(tree!=NULL)
+    {
+       int date=returnDate();
+        if(tree->Medicine.dateOfExpiration<date)
+        {
+            T=insertee(tree->Medicine.name,T);
+
+        }
+        Expired( tree->left, level + 1 );
+        Expired( tree->right, level + 1 );
+    }
+    return T;
+}
+void readInstructions(FILE *f,FILE *g,NodeT *tree)
+{    Pharmacy M;
+    M.name=(char*) malloc (sizeof(char)*MAXNAME);
+    char *strbuf=(char* ) malloc (sizeof(char)*MAXNAME);
+    while(fscanf(f,"%s",strbuf)!=EOF)
+    {
+        char *p;
+        p=strtok(strbuf,",\n");
+        if(p[0]=='i')
+        {
+            strcpy(p,p+1);
+            strcpy(M.name,p);
+            p=strtok(NULL,",\n");
+            M.price=atof(p);
+            p=strtok(NULL,",\n");
+            M.amount=atoi(p);
+            p=strtok(NULL,",\n");
+            M.dateOfManufacturing=atoi(p);
+            p=strtok(NULL,",\n");
+            M.dateOfExpiration=atoi(p);
+            tree=insert (M,tree);
+
+
+        }
+        else if(p[0]=='u')
+            {strcpy(p,p+1);
+             tree=find (p,tree);
+             p=strtok(NULL,",\n");
+            if(strcmp(p,"~")!=0)strcpy(tree->Medicine.name,p);
+            p=strtok(NULL,",\n");
+            if(strcmp(p,"~")!=0)tree->Medicine.price=atof(p);
+            p=strtok(NULL,",\n");
+            if(strcmp(p,"~")!=0)tree->Medicine.amount=atoi(p);
+             p=strtok(NULL,",\n");
+            if(strcmp(p,"~")!=0)tree->Medicine.dateOfManufacturing=atoi(p);
+            p=strtok(NULL,",\n");
+            if(strcmp(p,"~")!=0)tree->Medicine.dateOfExpiration=atoi(p);
+            }
+            else if(p[0]=='d')
+            {strcpy(p,p+1);
+             tree=deletee(p,tree);
+            }
+            else if(strcmp(p,"ce")==0)
+                 {
+                  NodeChar *Expire;
+                  Expire=Expired(tree,0);
+                  inorder(Expire,0,g);
+                 }
+
+    }
+}
+
+int main()
+{
+    FILE *f=fopen ("input.txt","r+");
+    FILE *g=fopen ("output.txt","r+");
+    NodeT *tree;
+    tree=NULL;
+    readInstructions(f,g,tree);
+    return 0;
+}
```

Comment:
awful formatting.. please use auto format from CodeBlocks.. your code is really hard to follow this way